client/src/pages/AdminPage/ManageSuggestionsPage/ManageSuggestionsPage.js

```diff
@@ -0,0 +1,152 @@
+    }
+
+    handleSuggestionApproveButtonClick() {
+        const { dispatch, match: { params: { suggestionId } }, location: { search } } = this.props;
+        const type = QueryString.parse(search).type;
+
+        if (suggestionId && type) {
+            try {
+                if (!isNaN(parseInt(suggestionId))) {
+                    switch (type) {
+                        case 'create':
+                            dispatch(approveCreateSuggestion(suggestionId));
+                            break;
+                        case 'update':
+                            dispatch(approveUpdateSuggestion(suggestionId));
+                            break;
+                        case 'bulk':
+                            dispatch(approveBulkCreateSuggestion(suggestionId));
+                            break;
+                        default:
+                            return;
+                    }
+                }
+            } catch (err) {}
+        }
+    }
+
+    handleSuggestionRejectButtonClick() {
+        const { dispatch, match: { params: { suggestionId } }, location: { search } } = this.props;
+        const type = QueryString.parse(search).type;
+
+        if (suggestionId && type) {
+            try {
+                if (!isNaN(parseInt(suggestionId))) {
+                    switch (type) {
+                        case 'create':
+                            dispatch(rejectCreateSuggestion(suggestionId));
+                            break;
+                        case 'update':
+                            dispatch(rejectUpdateSuggestion(suggestionId));
+                            break;
+                        case 'bulk':
+                            dispatch(rejectBulkCreateSuggestion(suggestionId));
+                            break;
+                        default:
+                            return;
+                    }
+                }
+            } catch (err) {}
+        }
+    }
+
+    render() {
+        const { mode, fetchCreateSuggestion, fetchUpdateSuggestion, fetchBulkCreateSuggestion,
+            approveCreateSuggestion, rejectCreateSuggestion, approveUpdateSuggestion, rejectUpdateSuggestion,
+            location: { search }, rejectBulkCreateSuggestion, approveBulkCreateSuggestion } = this.props;
+        const type = QueryString.parse(search).type;
+
+        const suggestion = approveCreateSuggestion.result || rejectCreateSuggestion.result ||
+            approveUpdateSuggestion.result || rejectUpdateSuggestion.result || rejectBulkCreateSuggestion.result ||
+            fetchCreateSuggestion.result || fetchUpdateSuggestion.result || fetchBulkCreateSuggestion.result;
+
+        const pending = approveCreateSuggestion.pending || rejectCreateSuggestion.pending ||
+            approveUpdateSuggestion.pending || rejectUpdateSuggestion.pending || rejectBulkCreateSuggestion.pending ||
+            fetchCreateSuggestion.pending || fetchUpdateSuggestion.pending || fetchBulkCreateSuggestion.pending;
```

Comment:
Again, really not the prettiest. I'm not sure if there's a better way to handle a Page component that is connected to a **bunch** of different Redux stores.